middle_end/flambda2/types/env/typing_env.ml

```diff
@@ -211,6 +212,21 @@ module Join_env = struct
     Meet_env.already_meeting central_env simple1 simple2
 end
 
+let names_to_types t =
+  Cached_level.names_to_types (One_level.just_after_level t.current_level)
+
+exception Binding_time_resolver_failure
+
+let binding_time_resolver resolver name =
+  match resolver (Name.compilation_unit name) with
```

Comment:
I think you should catch exceptions coming from the call to `resolver` and translate them to a fatal error, similar to what `find_with_binding_time_and_mode'` is doing (it helps with debugging)

Reply:
Done